### PR TITLE
Main Controller

### LINKED ISSUE #2691: legacy modules always use home realm in theme
I can use the dropdown options to change how Home and Admin look (1 column, 2 columns, 2 columns with centerblock, 3 columns, 3 columns with centerblock), but selecting these options makes no different when applied to Master.  This means, all my pages have the same layout as Home - in my case 3 columns with centerblock.

As I understand it, Master should control the display of all pages except Home and Admin.

### PR DESCRIPTION
**Pull request by craigh**
Use routing for all legacy urls. update themeEngine to use RequestStack. Correct computation of theme realm. Add startpage controller setting. Update Symfony developer toolbar styling and content.

| Q                 | A
| ----------------- | ---
| Bug fix?          | yes
| New feature?      | yes
| BC breaks?        | no
| Deprecations?     | no
| Tests pass?       | n/a
| Fixed tickets     | #2691, #2454
| Refs tickets      | -
| License           | MIT
| Doc PR            | n/a
| Changelog updated | yes

Fixes #2691, #2454.

### CODE REVIEW

**Comment by Guite** [comment]
This is for #2454 ?

**Comment by craigh** [comment]
> This is for #2454 ?
yes. It also should fix #2691 and it adds some features as well. And it's not finished.

**Comment by craigh** [comment]
It adds theme info to the developer's toolbar. It also fixes the styling in the dev toolbar to match Sy2.8

**Comment by craigh** [comment]
@cmfcmf @Guite @Kaik and anyone that cares... This branch is working quite well now from my testing. The basic idea is that it replaces the old zikula handling of "legacy" urls with symfony-style controllers/routes. This means that *all* zikula urls now will be _routes_. In my own testing this has a dramatic effect in speeding up the entire site as well as providing other benefits.

Anyway - it would be helpful if you could test this ASAP for me. You will need to rebuild your cache. Check and make sure all pages render properly for "old" modules (I tested News) that are 1.2.x or 1.3.x compatible and "new" modules (I tested Spec and system mods). Check both standard urls and shorturls. Please report issues asap. thanks!

File: src/lib/Zikula/Core/Theme/Engine.php [hunk 1]
```diff
@@ -323,39 +318,46 @@ public function positionIsAvailableInTheme($name)
...
 
             return;
         }
+        $request = $this->requestStack->getMasterRequest();
+        $requestAttributes = $request->attributes->all();
         // @todo BC remove at Core-2.0
-        if ((isset($this->requestAttributes['_zkType']) && $this->requestAttributes['_zkType'] == 'admin') || (isset($this->requestAttributes['lct']))) {
+        $lct = $request->query->get('lct', null);
+        if ((isset($requestAttributes['_zkType']) && $requestAttributes['_zkType'] == 'admin')
+            || ((isset($requestAttributes['_route']) && $requestAttributes['_route'] == 'legacy') && ($request->query->get('type') == 'admin'))
```
Comment (craigh) [inline src/lib/Zikula/Core/Theme/Engine.php]: ping @craigh add _route == legacy_shorturl

File: src/lib/Zikula/Bundle/CoreBundle/Resources/views/Collector/ZikulaVersion.html.twig [hunk 2]
```diff
@@ -19,6 +28,7 @@
     <div class="sf-toolbar-info-piece">
         <a href="{{ collector.ghZikulaBootstrapDocsUrl }}">Styling Documentation</a>
     </div>
+        {{ dump(collector.themeEngine) }}
```
Comment (Guite) [inline src/lib/Zikula/Bundle/CoreBundle/Resources/views/Collector/ZikulaVersion.html.twig]: This is probably to be removed.

File: src/system/SettingsModule/Resources/views/Admin/modifyconfig.tpl [hunk 3]
```diff
@@ -152,6 +152,16 @@
...
+                    <em class="help-block">{gt text="MyModuleName:Controller:method"}</em>
+                    <em class="help-block alert alert-warning">{gt text="Start Controller will take precedence over module/type/function settings below."}<br />
+                        {gt text="This setting only works with Core-1.4.x/2.0 compatible modules (with routes)."}<br />
+                        {gt text="Start function arguments (below) can also be combined with this setting."}</em>
+                </div>
+            </div>
+            <div class="form-group">
                 <label class="col-sm-3 control-label" for="settings_startpage">{gt text='Start module'}</label>
                 <div class="col-sm-9">
                     <select class="form-control" id="settings_startpage" name="settings[startpage]">
```
Comment (Guite) [inline src/system/SettingsModule/Resources/views/Admin/modifyconfig.tpl]: Maybe add another help-block about this setting is going to become deprecated?

**Pull request merged**